src/main/java/org/broadinstitute/hellbender/tools/spark/CreateIndexForHadoopBam.java

```diff
@@ -0,0 +1,158 @@
+            optional = true)
+    public int granularity = SplittingBAMIndexer.DEFAULT_GRANULARITY;
+
+    @Argument(fullName = CREATE_BAI_LONG_NAME,
+            shortName = CREATE_BAI_SHORT_NAME,
+            doc = "set this to create a bai index at the same time as creating a splitting index",
+            optional = true)
+    public boolean createBai = false;
+
+
+    @Override
+    protected Object doWork() {
+        if( granularity <= 0) {
+            throw new UserException.BadArgumentValue(SPLITTING_INDEX_GRANULARITY_LONG_NAME, Integer.toString(granularity), "Granularity must be > 0");
+        }
+        final File index = getOutputFile(output, inputBam);
+        if(createBai){
+            createBaiAndSplittingIndex(inputBam, index, granularity, readValidationStringency);
+        } else {
+            createOnlySplittingIndex(inputBam, index, granularity);
+        }
+
+        return 0;
+    }
+
+    private static void createOnlySplittingIndex(File inputBam, File index, int granularity) {
+        assertIsBam(inputBam);
+        //createBamSplittingIndex(inputBam, getOutputFile(output, inputBam), readValidationStringency, granularity);
+        try(BufferedInputStream in = new BufferedInputStream(new FileInputStream(inputBam))){
+            try(BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(index))) {
+                SplittingBAMIndexer.index(in, out, inputBam.length(), granularity);
+            } catch (IOException e){
+                throw new UserException.CouldNotCreateOutputFile(index, e);
+            }
+        } catch (IOException e) {
+            throw new UserException.CouldNotReadInputFile(inputBam, e);
+        }
+    }
+
+    private static void createBaiAndSplittingIndex(File inputBam, File index, int granularity, ValidationStringency readValidationStringency) {
+        assertIsBam(inputBam);
+        try(SamReader reader = SamReaderFactory.makeDefault()
+                .validationStringency(readValidationStringency)
+                .setOption(SamReaderFactory.Option.INCLUDE_SOURCE_IN_RECORDS, true)
+                .open(inputBam)){
+            try(BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(index))){
+                final SAMFileHeader header = reader.getFileHeader();
+                assertBamIsCoordinateSorted(header);
+                SplittingBAMIndexer indexer = new SplittingBAMIndexer(out, granularity);
+
+                BAMIndexer bamIndexer = new BAMIndexer(IOUtils.replaceExtension(index, BAMIndex.BAMIndexSuffix), header);
+                for(SAMRecord read : reader){
+                    indexer.processAlignment(read);
+                    bamIndexer.processAlignment(read);
+                }
+                indexer.finish(inputBam.length());
+                bamIndexer.finish();
+            } catch (final IOException e){
+                throw new UserException.CouldNotCreateOutputFile(index, e);
```

Comment:
same question - i dont think it's always true that using CouldNotCreateOutputFile is ok here